ha/test/integration/test_sigterm_to_ha_services.py

```diff
@@ -35,70 +40,105 @@ def start_all_services(self):
                                     shell=False,
                                     stdout=PIPE,
                                     stderr=PIPE)
         print(f"Started: health monitor, pid: {self.health_monitor.pid}")
         self.services = [self.k8s_monitor, self.fault_tolerance, self.health_monitor]
 
+    def send_start_cluster_shutdown_message(self):
+        """
+        Sends the "start_cluster_shutdown" message for cluster stop monitor
+        """
+        producer_id="csm_producer"
+        self.message_type = Conf.get(const.HA_GLOBAL_INDEX, f'CLUSTER_STOP_MON{const._DELIM}message_type')
+        self.producer = MessageBus.get_producer(producer_id=producer_id, message_type=self.message_type)
+        self.producer.publish({"start_cluster_shutdown":"1"})
+        # The consumer 'ClusterStopMonitor' will checking this message/key as cluster_alert["start_cluster_shutdown"] == '1'
+
+    def check_cluster_stop_key(self, should_exists):
+        """
+        This checks if const.CLUSTER_STOP_KEY is exist or not in confsore and assert on failure
+        should_exists : if it is true and key doen not exist then assert (it raise failure exception) and vice versa.
+        This key should be added by cluster stop monitor (falult tolerance monitor service) and
+        should be get deleted by k8s monitor after receiving sigterm.
+        """
+        key_exists = self.confstore.key_exists(const.CLUSTER_STOP_KEY)
+        self.assertTrue(key_exists == should_exists, f"Error: The Key {const.CLUSTER_STOP_KEY} must should be deleted by k8s monitor only after sigterm is received.")
+
+    def cleanup_cluster_stop(self):
+        """
+        Cleanup the key const.CLUSTER_STOP_KEY from confstore.
+        """
+        if self.confstore.key_exists(const.CLUSTER_STOP_KEY):
+            print(f"The Key {const.CLUSTER_STOP_KEY} is exist in confstore so deleting now.")
+            self.confstore.delete(const.CLUSTER_STOP_KEY)
 
     def send_sigterm(self):
         """
         Send sigterm (15) signal to all services
         """
-        print("sending sigterm to all services.")
+        print("Sending SIGTERM to all services.")
         for service in self.services:
             service.send_signal(15)
 
-    def check_for_stop(self, wait_time=20):
+    def check_running_services(self):
+        """
+        Check if all services are running if not then assert
+        """
+        services = [service for service in self.services]
+
+        # filter out the running services
+        print("Check running services.")
+        services[:] = filterfalse(lambda service : service.poll() != None, services)
+        print(f"Running service: {[service.args for service in services]}")
+        self.assertTrue(len(services) == len(self.services), "Not all HA services are running.")
+
+    def check_for_stop(self, wait_time=30) -> int:
```

Comment:
The default of wait_time is fine for now. Eventually can be initialized in miniprovisioning (need to be made available by provisioning, picked up from services yaml / conf global settings of grace period) in ha confstore and used.

Reply:
Please resolve and close this comment if it is not present in current cluster.conf.